src/components/app.tsx

```diff
@@ -66,67 +116,35 @@ export const App = () => {
     setPlaybackProgress(0);
   };
 
-  const setupAudioContext = async (soundName: SoundName) => {
-    if (audioSource.current && audioContext.current) {
-      audioSource.current.stop();
-      await audioContext.current.close();
-      setPlaying(false);
-    }
-
-    // When a user chooses to record their own sound, we don't start recording
-    // immediately. But we do want to clear out the old sound data here, and to
-    // update the playback progress indicator, so that it is clear that there
-    // is nothing recorded (yet).
-    if (soundName === "record-my-own") {
-      const emptyBuffer = new AudioBuffer({
-        length: 1,
-        sampleRate: SOUND_SAMPLE_RATE
-      });
-      audioContext.current = new AudioContext();
-      gainNode.current = audioContext.current.createGain();
-      setAudioBuffer(emptyBuffer);
-      setPlaybackProgress(0);
-      return;
-    }
-
-    // Handle selection of 'canned' sounds...
-    const response = await window.fetch(sounds[soundName]);
-    const soundArrayBuffer = await response.arrayBuffer();
-    audioContext.current = new AudioContext();
-    gainNode.current = audioContext.current.createGain();
-
-    setAudioBuffer(await audioContext.current.decodeAudioData(soundArrayBuffer));
-    setPlaybackProgress(0);
-  };
+  const setupAudioContextCallback = useCallback((
+    soundName) => {
+        setupAudioContext(audioSource, audioContext, setPlaying, gainNode, setAudioBuffer, recordingAudioBuffer, setPlaybackProgress, soundName);
```

Comment:
The indirection through the module-level `setupAudioContext()` function is unnecessary.